### DISCUSSION TITLE
Example: Using Wokwi (MCU Online Simulator) on Pico

### DISCUSSION THREAD

**Discussion by madias123** [earlephilhower/arduino-pico#1547] [Ideas]
**Wokwi** - (not only) Arduino Simulator
Since Wokwi also supports the RP2040 (Pi Pico in this case), I made a default preset with the Earle F. Philhower, III core enabled.
Link:   [https://wokwi.com/projects/367880033507346433](url)
Danger! Always use "Serial1" instead of "Serial".
Background: In order to report bugs, for example, the user could (if possible) rebuild a simulation online, since many "standard parts" are available and drop a wokwi link.

**Playing with PIO - Code:**
A great possibility (that's what I came to the site for in the first place) is also the convenient online conversation of the PIO code to "C SDK":
https://wokwi.com/tools/pioasm
Of course, this can also be simulated immediately, e.g. I played with a SDK quadrature encoder example:
https://wokwi.com/projects/367881190996882433
(unfortunately runs a bit slow)
Faster example with PIO encoders would be here:
https://wokwi.com/projects/367881190996882433

**PIO Debugger**
After hitting "Run" you can also select the PIO debugger - it's a little bit hidden and appears next to all used files AFTER hitting run.
![image](https://github.com/earlephilhower/arduino-pico/assets/3707465/395bb294-2ea4-4115-8e6d-6b0af2930b93)

**Comment by earlephilhower** [comment]
Very nice, thanks for the linkage!
